### PR TITLE
GAS: update auth provider docs

### PR DESCRIPTION
**Pull request by IevaVasiljeva**
**What is this feature?**

Updates auth provider docs to reference GAS.

**Which issue(s) does this PR fix?**:

Related to https://github.com/grafana/identity-access-team/issues/935

**Special notes for your reviewer:**

This PR builds on top of https://github.com/grafana/grafana/pull/96090.

I think it's a useful change, as it makes GAS easier to configure, because people can refer to provider specific docs for configuration details. But I'm a bit weary merging it now, in case we end up not GA'ing GAS, as it's a larger change to revert.

### CODE REVIEW

File: docs/sources/setup-grafana/configure-security/configure-authentication/azuread/index.md [hunk 1]
```diff
@@ -345,14 +345,14 @@ This setting is ignored if multiple auth providers are configured to use auto lo
 auto_login = true
 ```
 
-### Team Sync (Enterprise only)
+### Group sync (Enterprise only)
 
-With Team Sync you can map your Entra ID groups to teams in Grafana so that your users will automatically be added to
-the correct teams.
+With group sync you can map your Entra ID groups to teams and role-based access control roles in Grafana. This allows users to automatically be added to
```
Comment (eleijonmarck) [inline docs/sources/setup-grafana/configure-security/configure-authentication/azuread/index.md:350]:
```suggestion
With group sync you can map your Entra ID groups to teams and roles in Grafana. This allows users to automatically be added to
```

Comment (eleijonmarck) [reply docs/sources/setup-grafana/configure-security/configure-authentication/azuread/index.md:350]: not needed to mention role based access control here i think

File: docs/sources/setup-grafana/configure-security/configure-authentication/enhanced-ldap/index.md [hunk 2]
```diff
@@ -28,18 +28,16 @@ The enhanced LDAP integration adds additional functionality on top of the [LDAP
 
 > To control user access with role-based permissions, refer to [role-based access control]({{< relref "../../../../administration/roles-and-permissions/access-control" >}}).
 
-## LDAP group synchronization for teams
+## LDAP group synchronization
 
-With enhanced LDAP integration, you can set up synchronization between LDAP groups and teams. This enables LDAP users that are members
-of certain LDAP groups to automatically be added or removed as members to certain teams in Grafana.
+With enhanced LDAP integration, you can set up synchronization between LDAP groups and Grafana teams and role-based access control roles. This enables LDAP users that are members
+of certain LDAP groups to be automatically added to teams and granted roles in Grafana.
```
Comment (eleijonmarck) [inline docs/sources/setup-grafana/configure-security/configure-authentication/enhanced-ldap/index.md:34]: nit: i do not know if i am reading this incorrectly. The wording of `to team and granted roles` seems like it is only specific roles. Is this correct?

```suggestion
of certain LDAP groups to be automatically added to teams and granted roles in Grafana.
```

File: docs/sources/setup-grafana/configure-security/configure-authentication/generic-oauth/index.md [hunk 3]
```diff
@@ -328,19 +328,21 @@ org_attribute_path = info.roles
...
+## Configure group synchronization
 
-> **Note:** Available in [Grafana Enterprise]({{< relref "../../../../introduction/grafana-enterprise" >}}) and [Grafana Cloud](/docs/grafana-cloud/).
+{{< admonition type="note" >}}
+Available in [Grafana Enterprise](https://grafana.com/docs/grafana/<GRAFANA_VERSION>/introduction/grafana-enterprise) and [Grafana Cloud](/docs/grafana-cloud/).
+{{< /admonition >}}
 
-By using Team Sync, you can link your OAuth2 groups to teams within Grafana. This will automatically assign users to the appropriate teams.
-Teams for each user are synchronized when the user logs in.
+Grafana supports synchronization of OAuth2 groups with Grafana teams and role-based access control roles. This allows automatically assigning users to the appropriate teams or automatically granting them the mapped roles.
```
Comment (eleijonmarck) [inline docs/sources/setup-grafana/configure-security/configure-authentication/generic-oauth/index.md:337]: > assigning users to the appropriate teams or automatically granting them the mapped roles.

is the feature different for each provider? in that we can map to teams or roles. or is it that we can map to teams and roles as well here.

**Review by irenerl24 — APPROVED** [review]
LGTM from a docs perspective :)

File: docs/sources/setup-grafana/configure-security/configure-authentication/generic-oauth/index.md (hunk 3 above)
Comment (IevaVasiljeva) [reply docs/sources/setup-grafana/configure-security/configure-authentication/generic-oauth/index.md:337]: The latter - it's not different between providers, but we do have provider specific setup guides, so I've appended this info to all of them.

File: docs/sources/setup-grafana/configure-security/configure-authentication/enhanced-ldap/index.md (hunk 2 above)
Comment (IevaVasiljeva) [reply docs/sources/setup-grafana/configure-security/configure-authentication/enhanced-ldap/index.md:34]: Thanks for the feedback, I'll try to reword this.

**Pull request merged**